l1-contracts/src.ts/deploy.ts

```diff
@@ -783,18 +805,37 @@ export class Deployer {
     }
 
     const diamondProxy = this.stateTransitionContract(this.deployWallet);
-    const tx4 = await diamondProxy.setTokenMultiplier(1, 1);
-    const receipt4 = await tx4.wait();
-    if (this.verbose) {
-      console.log(`BaseTokenMultiplier set, gas used: ${receipt4.gasUsed.toString()}`);
-    }
-
-    if (validiumMode) {
-      const tx5 = await diamondProxy.setPubdataPricingMode(PubdataPricingMode.Validium);
-      const receipt5 = await tx5.wait();
+    // if we are using governance, the deployer will not be the admin, so we can't call the diamond proxy directly
+    if (admin == this.deployWallet.address) {
+      const tx4 = await diamondProxy.setTokenMultiplier(1, 1);
+      const receipt4 = await tx4.wait();
       if (this.verbose) {
-        console.log(`Validium mode set, gas used: ${receipt5.gasUsed.toString()}`);
+        console.log(`BaseTokenMultiplier set, gas used: ${receipt4.gasUsed.toString()}`);
+      }
+
+      if (validiumMode) {
+        const tx5 = await diamondProxy.setPubdataPricingMode(PubdataPricingMode.Validium);
+        const receipt5 = await tx5.wait();
+        if (this.verbose) {
+          console.log(`Validium mode set, gas used: ${receipt5.gasUsed.toString()}`);
+        }
       }
+    } else if (useGovernance) {
```

Comment:
just in case, we shouldnt typically do it for custom chains, i.e. change their multiplier through STM, why do we need this functionality?

Reply:
my motivation was that the multiplier is required to run the chain, and it's not being set by default (AFAIK), so it has to be set somehow. So if we have the access to the governance in the script - we can do that to remove 1 logical step from the partners (even though they can do it themselves using the admin account)

Reply:
Could you please use executeDirectOrGovernance ?

Reply:
I don't think so, executeDirectOrGovernance calls the same contract either directly or through governance, but in this case we call the STM through the governance or the Diamond directly

Reply:
Could you not call the STM in both cases?

Reply:
nope, STM's function has onlyOwner modifier

Reply:
@dimazhornyk the owner of the STM is the `Governance` contract controlled by multisig. We will not sign this message every single time a chain is added. This should be done by the admin of the chain

Reply:
got it, removed that call